rust/garbage_collector/src/garbage_collector_orchestrator_v2.rs

```diff
@@ -213,6 +215,55 @@ impl Orchestrator for GarbageCollectorOrchestrator {
 }
 
 impl GarbageCollectorOrchestrator {
+    async fn get_eligible_soft_deleted_collections_to_gc(
+        &mut self,
+        all_collection_ids: Vec<CollectionUuid>,
+    ) -> Result<HashSet<CollectionUuid>, GarbageCollectorError> {
+        let soft_delete_statuses = self
+            .sysdb_client
+            .batch_get_collection_soft_delete_status(all_collection_ids.clone())
+            .await
```

Comment:
seems like this api is a subset of `sysdb.get_collections()`? It only returns the soft delete status whereas the `get_collections()` call returns the soft delete status as well as other info?